staging/src/k8s.io/apimachinery/pkg/api/resource/quantity_test.go

```diff
@@ -736,15 +736,128 @@ func TestQuantityParseEmit(t *testing.T) {
 		in     string
 		expect string
 	}{
+
+		// k (>1)
+		{"1k", "1k"},
+		{"1.23k", "1230"},
+		{"1024k", "1024k"},
+		{"1224k", "1224k"},
+		{"1536k", "1536k"},
+		{"1836k", "1836k"},
+
+		// k (<1)
+		{".2k", "200"},
+		{".5k", "500"},
+		{".02k", "20"},
+		{".023k", "23"},
+		{".0234k", "23400m"},
+		{".0004k", "400m"},
+		{".001k", "1"},
+		{".000001k", "1m"},
+		{".000000001k", "1u"},
+		{".000000000001k", "1n"},
+
+		// Ki (>1)
 		{"1Ki", "1Ki"},
+		{"1.23Ki", "1.23Ki"},
+		{"1024Ki", "1Mi"},
+		{"1224Ki", "1224Ki"},
+		{"1536Ki", "1536Ki"},
+		{"1836Ki", "1836Ki"},
+
+		// Ki (<1)
+		{".2Ki", ".2Ki"},
+		{".5Ki", ".5Ki"},
+		{".02Ki", ".02Ki"},
+		{".023Ki", ".023Ki"},
+		{".0234Ki", ".0234Ki"},
+		{".0004Ki", ".0004Ki"},
+		{".001Ki", ".001Ki"},
+		{".000001Ki", ".000001Ki"},
+		{".000000001Ki", ".000000001Ki"},
+		{".000000000001Ki", "2n"},
+
+		// M (>1)
+		{"1M", "1M"},
+		{"1.23M", "1230k"},
+		{"1024M", "1024M"},
+		{"1224M", "1224M"},
+		{"1536M", "1536M"},
+		{"1836M", "1836M"},
+
+		// M (<1)
+		{".2M", "200k"},
+		{".5M", "500k"},
+		{".02M", "20k"},
+		{".023M", "23k"},
+		{".0234M", "23400"},
+		{".0004M", "400"},
+		{".001M", "1k"},
+		{".000001M", "1"},
+		{".000000001M", "1m"},
+		{".000000000001M", "1u"},
+
+		// Mi (>1)
 		{"1Mi", "1Mi"},
-		{"1Gi", "1Gi"},
+		{"1.23Mi", "1.23Mi"},
 		{"1024Mi", "1Gi"},
-		{"1000M", "1G"},
```

Comment:
I have added these numbers back, not sure how I missed them. If you have any other numbers in mind, let me know.